### DISCUSSION TITLE
Device Sync Deprecation

### DISCUSSION THREAD

**Discussion by jamalsoueidan** [realm/realm-js#6884] [General]
Is Realm project going to be deprecated or what exactly? 
Please let us know! @kraenhansen  @kneth

**Comment by kraenhansen** [comment]
Today, we announced the deprecation of Atlas Device Sync + Realm SDKs in September 2024. For more information please see:
 - [SDK Deprecation](https://www.mongodb.com/docs/atlas/device-sdks/device-sdk-deprecation)
 - [Device Sync Deprecation](https://www.mongodb.com/docs/atlas/app-services/sync/device-sync-deprecation)

For a version of RealmJS without sync features, install from [community on npm](https://www.npmjs.com/package/realm/v/community) or see the `community` git branch.

**Reply by wesvance** [reply]
Any chance for the Sync to become open source / community driven? This is a big hit to all of us who have built applications on top of sync.

**Reply by mpobrien** [reply]
Releasing the code for the cloud layer is unfortunately not an option right now, since it's deeply intertwined with the control plane still in production, which would have security implications. However the legacy server code remains open source at https://github.com/realm/realm-core/tree/master/src/realm/sync/noinst/server.

**Comment by GitMurf** [comment]
Do we know who will be the "owner" of the community version going forward? Is the current core team all still staying and working for Mongo on other projects? Will Mongo we supporting the further development / maintenance of Realm in the future when it is a community / fully open source project? Will some of the core team from Mongo we leaving to maintain / continue to develop Realm?

Basically I need to understand what the viability and long term success of the core Realm (local) product will be before I even need to consider figuring out an alternative sync solution. If the project is going to be completely open source and led by the community without the current core team involved then we need to understand that because that is a deal breaker and we will need to move off of Realm immediately.

Thanks in advance for any further context / information you can provide.

**Comment by rhenandias** [comment]
I was about to start a project now using Realm, how does this affect new projects?

I will still be able to use Realm as a local database, right? However, I won't be able to sync the information with MongoDB databases hosted on MongoDB's own platforms (Atlas Device Sync), is that correct?

If I want to use Realm but build my own backend with a MongoDB database, is that possible?

**Reply by kraenhansen** [reply]
> is that correct?

Right. I would add that a large part of the maintenance that the community would have to handle now is keeping up with the React Native community. You'd be in the mercy of other Realm users to keep the SDK updated as React Native inevitably change in ways that the SDK needs to adopt to.

> If I want to use Realm but build my own backend with a MongoDB database, is that possible?

As with anything in software, nothing is impossible, but a fully generic sync server isn't trivial to implement.

**Reply by liamjones** [reply]
> I would add that a large part of the maintenance that the community would have to handle now is keeping up with the React Native community. You'd be in the mercy of other Realm users to keep the SDK updated as React Native inevitably change in ways that the SDK needs to adopt to.

@kraenhansen Something I can't see clarity on in any of the announcement pages; are MongoDB ceasing internal development on the non-sync Realm SDKs? I'd say it's implied by the deprecation announcement & the `community` branch name but there doesn't seem to be an explicit statement anywhere I can see.

If so, how are PR merges to the community branch going to work?

**Reply by kraenhansen** [reply]
@liamjones good question. I'll try to clarify.

> are MongoDB ceasing internal development on the non-sync Realm SDKs

To be clear, we're no longer doing active feature development on any of the SDKs: We're mostly in a "keep the lights on" mode for existing customers until everything goes EOL in less than a year from now. This goes for sync as well as non-sync parts of the product.

> how are PR merges to the community branch going to work?

This is a good question and to be honest, as it involves speculation about the future, there's a limit to how accurately I'm able to answer it. To be honest, we haven't seen a large engagement from the community to provide PRs and in the current (low) rate I'd expect we'll (MongoDB engineers now assigned other teams within the company) be able to handle reviewing, merging and releasing for the foreseeable future.

If however, someone from the community shows an interest in participating in the release and ongoing maintenance of the SDK, that would be welcomed and we'd be interested in talking through what that would look like in practice and any mutual expectations there might be.

**Reply by liamjones** [reply]
Thank you for the reply @kraenhansen.

I'll be sad to see Realm go if the community can't keep it going after the EOL. Thanks to you & the rest of the team who worked on it all these years. ❤️ 

One more query that came to mind after your reply; I understand that the device SDKs are likely to need updates as RN/iOS/Android moves forward. Is realm-core affected by these upgrades? Or largely not because it's C++ and has realm-js, realm-swift, etc between it and the device OS?

**Reply by kraenhansen** [reply]
> Thanks to you & the rest of the team

Thanks 🙂 We've thoroughly enjoyed it.

> Is realm-core affected by these upgrades?

In most cases these updates are primarily SDK specific and since https://github.com/realm/realm-js/pull/6650 we're driving all the building of Realm Core from the SDK repository too. There are some changes to the platforms that do require updates to the Core codebase ([support for 16 KB page sizes on Android](https://developer.android.com/guide/practices/page-sizes) comes to mind).

**Comment by luangarcia** [comment]
How do we stand with Device Sync services? Are we going to get a Realm Object Server (ROS) replacement? Or are we being abandoned by MongoDB?

This decision heavily impacts projects running in production, and we are waiting for a clear response from MongoDB regarding these services. The official page suggests migrating, but there are no compatible services with React Native that offer the same level of integration for schema-based syncing with MongoDB as Device Sync did.

What should we do at this point? Should we start rewriting entire projects, or is it better to wait for the community? We need more clarification to plan our next steps.

**Reply by GitMurf** [reply]
I hate to say it but all indications I have gotten and seen (and even from this thread you can see no real engagement from the Mongo team)... the reality is much of the team (as I have been told) has already been re-orged to other teams inside Mongo.

I would not count on anything other than being able to use Realm locally as it sits today (no confidence in substantial updates), and surely no real sync option provided by Mongo. You will likely have to roll your own.

We have already begun migrating our app and re-writing our schema to use SQLite. It is extremely frustrating that Mongo left us all hanging like this. We certainly cannot trust our business with them moving forward, even if they did provide some potential options down the road (which is unlikely).
